smarts/core/sensors.py

```diff
@@ -318,6 +319,40 @@ def observe(sim, agent_id, sensor_state, vehicle):
     def step(sim, sensor_state):
         return sensor_state.step()
 
+    @classmethod
+    def _agents_alive_done_check(
+        cls, agent_manager, agents_alive: AgentsAliveDoneCriteria
+    ):
+        if not agents_alive:
+            return False
+
+        done = False
+        if agents_alive.total_ego_agents_alive:
+            done = (
+                len(agent_manager.ego_agent_ids) < agents_alive.total_ego_agents_alive
+            )
```

Comment:
Here (and in the following) you could just do:
 ```python
if agents_alive.total_ego_agents_alive and len(agent_manager.ego_agent_ids) < agents_alive.total_ego_agents_alive:
    return True
```
(rather than having multiple `if not done...` clauses.)

Reply:
I see. yea that makes the code simpler, and even removes the need to have a `done` flag